### PR TITLE
Fixed win screen,  new game button and got my new game function to work again

### PR DESCRIPTION
**Pull request by prra-codes**
Summary: Created useEffect to fix the win screens and new game button. Adjusted my newGame function to make an axios post request, and the request on the server resets the boardArr and currentColumn variables on the server side. The useEffect that runs on the client side every 0.5 seconds will then update the client side board and currentColumns states.

Test Plan: Ran npm start and npm run dev. Things are looking good!

### CODE REVIEW

File: client/src/components/Board/Board.js [hunk 1]
```diff
@@ -64,9 +68,30 @@ const Board = () => {
...
+      [null, null, null, null, null, null, null],
+    ];
+
+    const resetCurrentColumns = [5, 5, 5, 5, 5, 5, 5];
+    let player = winner === RED ? YELLOW : RED;
+
+    await axios.post("http://localhost:8000/newGame", {
+      newBoard,
+      resetCurrentColumns,
+      player,
```
Comment (jacksongabbard) [inline client/src/components/Board/Board.js]: Why do you need to send this info to the server? If you know you want to create a newGame, you should be able to do 100% of the setup on the server without the client passing specific game state data to the server.

File: server/server.js [hunk 2]
```diff
@@ -40,13 +35,35 @@ app.post("/setPiece", async (req, res) => {
...
+  const resetCurrentColumns = newGameData["resetCurrentColumns"];
+
+  currentColumns = [...resetCurrentColumns];
+
+  const resetGameBoard = newGameData["newBoard"];
+  boardArr = [...resetGameBoard];
+
+  const player = newGameData["player"];
+
+  currentPlayer = player;
```
Comment (jacksongabbard) [inline server/server.js]: This overall change is good, except this part. You shouldn't accept the entire game state blindly from the user. Suppose they send you a game state that is an array with 1,000,000 rows and a 1,000,000 columns to see if they can crash your server?

**Review by jacksongabbard — CHANGES_REQUESTED** [review]
You need to change the way the new game functionality works.

Also, you need to update the title and the description of the pull request to describe what it actually is now. "dealing with I believe to be async issues" isn't a very useful title for future you to make sense of.

File: client/src/components/Board/Board.js (hunk 1 above)
Comment (prra-codes) [reply client/src/components/Board/Board.js]: Understood.

File: server/server.js (hunk 2 above)
Comment (prra-codes) [reply server/server.js]: I'll update it. Thank you.

File: server/server.js [hunk 3]
```diff
@@ -40,13 +35,36 @@ app.post("/setPiece", async (req, res) => {
...
+    [null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null],
+  ];
+
+  const resetCurrentColumns = [5, 5, 5, 5, 5, 5, 5];
+
+  currentColumns = [...resetCurrentColumns];
+
+  boardArr = [...newBoard];
+
```
Comment (jacksongabbard) [inline server/server.js:65]: Do you need to reset who the current player is?

**Review by jacksongabbard — APPROVED** [review]
One question down below, but otherwise this looks good to me.

File: server/server.js (hunk 3 above)
Comment (prra-codes) [reply server/server.js:65]: No, when a player wins, the current player is already changed due to the setPiece function on the client and server.

E.g. If it's turn red, and red makes the winning move, the cursor will turn yellow and the board game will say "RED WINS!". So when the new game button is clicked, the cursor is already yellow.

**Comment by prra-codes** [comment]
Thank you!

No, when a player wins, the current player is already changed due to the
setPiece function on the client and server.

E.g. If it's turn red, and red makes the winning move, the cursor will turn
yellow and the board game will say "RED WINS!". So when the new game button
is clicked, the cursor is already yellow.

**Pull request merged**